packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -141,15 +141,19 @@ export async function prefetchLatestSnapshot(
 					});
 					assert(cacheP !== undefined, 0x1e7 /* "caching was not performed!" */);
 					await cacheP;
+					// Schedule it to remove from cache after 5s, so that if this snapshot is going to be used on load,
+					// and we don't find it in this cache, then we will make a network call again to fetch this
+					// snapshot which is redundant. So don't clear this cache immediately. We also clear this cache,
+					// if this cached snapshot is used on load.
+					setTimeout(() => {
+						snapshotNonPersistentCache?.remove(nonPersistentCacheKey);
+					}, 5000);
```

Comment:
Given that we remove this same cache line in fetchSnapshotCore(), should this number be bigger (like 10 seconds)?
I.e. we expect that in all cases if load follows (that is 99% case), then we use it and delete outside of the timer flow. And Making this longer ensures more sessions would hit it?

On other side, we do not cancel this timer on fetchSnapshotCore() (it would be hard to do), thus it could theoretically run into problems with multiple load sequences, so maybe keeping it low is beneficial? Not sure.

Reply:
I think we can be data driven here in immediate term and see how 5s works out. I don't expect many containers to load from same snapshot and I also don't expect apps to make prefetch call like more than 5s ago before they load the container. So for now 5s seems fine to me.